app/forms/idv/api_image_upload_form.rb

```diff
@@ -203,6 +203,7 @@ def track_event(event, attributes = {})
     def update_analytics(client_response)
       add_costs(client_response)
       update_funnel(client_response)
+      save_proofing_components
```

Comment:
I'm fine with this approach and this does seem to fit in what I might consider "analytics", but as an alternative, I notice that in other places we add proofing components, it's at the time where we call `extract_pii_from_doc`. By calling it in the [step](https://github.com/18F/identity-idp/blob/d3de31e64e890f358a221f075f5e5f2f6e59024c/app/services/idv/steps/document_capture_step.rb#L99) when we handle the stored result, we could avoid having to reimplement `save_proofing_components`, and maybe even refactor a bit to combine parts of `post_images_and_handle_result` and `handle_stored_result`.

Reply:
And actually, looking at this code, I wonder if this implementation could cause duplicate proofing components to be added if it's handled both by the hybrid-mobile API call, and then also by the hybrid-desktop's redirect?

https://github.com/18F/identity-idp/blob/d3de31e64e890f358a221f075f5e5f2f6e59024c/app/services/idv/steps/link_sent_step.rb#L19-L21

Reply:
So proofing components is a misleading name, there's a unique index so a user only has one `proofing_component` row at a time, so updating in place with the same data should be ok?

Reply:
> So proofing components is a misleading name, there's a unique index so a user only has one `proofing_component` row at a time, so updating in place with the same data should be ok?

Ah, okay. I didn't dig too deep, but had seen in the changes below that we "add" (`Db::ProofingComponent::Add`) a proofing component, so it wasn't immediately obvious if uniqueness was considered.

Reply:
Yeah I just broke out this PR to clean that up, take a quick look and it might explain some of the model more: https://github.com/18F/identity-idp/pull/5737


But to your original point, yes I'll look at seeing if I can re-implement it in the step to avoid duplication

Reply:
Much simplier in 820ae852279ec1d6b29df8599a0c99b70571b67c ! thanks for the reminder